docs/design/w3c/spec.md

```diff
@@ -0,0 +1,530 @@
+pointing to the difference in a base credential structure and looks the following:
+
+```
+{
+  ... 
+  "type": [
+    "VerifiableCredential",         // general verifiable credential definition
+    "AnonCredsCredential",          // definition for AnonCreds credentials
+  ]
+  ...
+}
+```
+
+#### Credential Subject
+
+W3C [Credential Subject](https://www.w3.org/TR/vc-data-model/#credential-subject) section requires including
+of `credentialSubject` property to verifiable credential.
+
+Credential subject value contains [claims](https://www.w3.org/TR/vc-data-model/#claims) about one or more subjects.
+
+In the context of W3C Announced credentials, credential subject property is compliant with the following statements:
+
+- credentials will always include claims about only one subjects.
+    - So that `credentialSubject` property will always be represented as an object entry, but not an array.
+- credentials claims are always represented as key value pairs, where `value` is the `raw` value of CL credential
+  attributes.
+    - encoded CL credential values are not included in the credential subject
+
+In the case of W3C AnonCreds credentials, the `credentialSubject` attribute looks the following:
+
+```
+{
+  ... 
+  "credentialSubject": {
+    "name": "Alice Jones",
+  }
+  ...
+}
+```
+
+* **TO DISCUSS**: W3C data model allows attributes to be represented not only as key/value string pairs but also as
+  objects and arrays.
+    * If we want to support more complex representations for the W3C AnonCreds credential attributes and their
+      presentations, we need to design following things:
+        * how encoding will work for such attributes
+        * how selective disclosure will work on top level attribute itself
+            ```
+              "credentialSubject": {
+                "address": {
+                    "type": "Address",
+                    "city": "Foo",
+                    "street": "Main str."
+                }
+              }
+            ```
+
+#### Data Schemas
+
+W3C [Credential Subject](https://www.w3.org/TR/vc-data-model/#data-schemas) section defines an optional capability to
+include `credentialSchema` property to enforce a specific structure on a given verifiable credential and encoding used
+to map the claims of a verifiable credential to an alternative representation format.
+
+In the context of W3C AnonCreds credentials defined a custom `AnonCredsDefinition` data schema in order to include the
+following information to credential:
+
+In the case of W3C AnonCreds credentials, the `credentialSchema` attribute defines a custom `AnonCredsDefinition`
+schema in order to include the information about Indy related definitions to credential and looks the following:
+
+```
+{
+  ... 
+  "credentialSchema": {
+    "type": "AnonCredsDefinition",
+    "definition": "did:sov:3avoBCqDMFHFaKUHug9s8W:3:CL:13:default",
+    "schema": "did:sov:3avoBCqDMFHFaKUHug9s8W:2:fabername:0.1.0",
+    "encoding": "auto"
```

Comment:
Would be good to discuss if we would ever support another encoding scheme. If that is not the case, we can just omit this value.

Reply:
For AnonCreds 1.0, we have no plans to support any other encoding scheme and an update to the spec. in that area is extremely unlikely.

Converting to `”encoding": “auto"` is a change, in that we can take the encoding out of the hands of the issuer and do it in the library.  I’m a fan of that, but I think we should leave in the `encoding`, since it allows for us to stick to the current implementation — e.g., putting in what the issuer provided.

```
  ... 
  “encoding": {
    "name": “12341241535",
  }
  ...

```

Definitely worth a discussion — even to the point of in the legacy AnonCreds format allowing for an `”encoding": “auto”`option where the issuer-provided encoding is ignored and or/rejected if inaccurate.